### DISCUSSION TITLE
Password validation for admin api

### DISCUSSION THREAD

**Discussion by neubami94** [ory/kratos#4050] [Ideas]
Hello, I would like to ask if there is a possibility for validating a password with HiBP API when it is being set with admin api? In our use case we need to set the password with admin api, instead of frontend api, but the password is not being validated there. What are our options? Would it be possible, to have an option to turn it on when adding or editing an identity? Or at least a separate endpoint for password validation? 

Thanks.

**Comment by vinckr** [comment]
Hello @neubami94 

You are creating users through the API right? 
Since I assume you have the password in plaintext available you can send it to HiBP directly to validate as part of the creation (or rather before). 
They already offer a ready made API for this, so IMO its most effective to just use that- or maybe I am missing something?

**Reply by neubami94** [reply]
Hello, 
yes I looked into that, however the HiBP API is a paid service. Our company would like to avoid paying for both ORY and HiBP when ORY itself already has password validation function (just not in the admin api).
